packages/asset-swapper/src/utils/market_operation_utils/fills.ts

```diff
@@ -155,6 +155,22 @@ function dexQuotesToPaths(
     return paths;
 }
 
+export function getTwoHopAdjustedRate(
+    side: MarketOperation,
+    twoHopQuote: DexSample<MultiHopFillData>,
+    targetInput: BigNumber,
+    ethToOutputRate: BigNumber,
+    fees: FeeSchedule = {},
+): BigNumber {
+    const { output, input, fillData } = twoHopQuote;
+    if (input.isLessThan(targetInput) || output.isZero()) {
+        return ZERO_AMOUNT;
+    }
+    const penalty = ethToOutputRate.times(fees[ERC20BridgeSource.MultiHop]!(fillData));
```

Comment:
Why isn't the penalty just the sources in the hop combined?

Reply:
o that's what I had it as initially, but I moved it over to the fee schedule for consistency with the other sources –– I also figure there's a chance we mess with it, e.g. apply the gas limit buffer multiplier to the "non-deterministic" hops